### PR TITLE
Bootstrap wrapper improvements 

### PR DESCRIPTION
**Pull request by Starzu**

### CODE REVIEW

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/form/UdashForm.scala [hunk 1]
```diff
@@ -82,11 +84,12 @@ object UdashForm {
     * More: <a href="http://getbootstrap.com/css/#forms-inline">Bootstrap Docs</a>.
     *
     * @param componentId Id of the root DOM node.
+    * @param onSubmit Callback executed when a form is submitted
     * @param content Form content
     * @return `UdashForm` component, call render to create DOM element.
     */
-  def inline(componentId: ComponentId, content: Modifier*): UdashForm =
-    new UdashForm(Some(BootstrapStyles.Form.formInline), componentId)(content)
+  def inline(componentId: ComponentId, onSubmit: Event => Any, content: Modifier*): UdashForm =
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/form/UdashForm.scala]: What do you think about moving `onSubmit` to a separate argument list? It feels more natural, a vararg can be confusing enough for a single list.

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/form/UdashForm.scala [hunk 2]
```diff
@@ -156,8 +160,9 @@ object UdashForm {
     inputGroup(inputId, validation)(labelContent)(FileInput(name, acceptMultipleFiles, selectedFiles)((id := inputId) +: input))
 
   /** Creates checkbox. */
-  def checkbox(validation: Option[Modifier] = None)(labelContent: Modifier*)(property: Property[Boolean], input: Modifier*): Modifier =
-    div(BootstrapStyles.Form.checkbox)(
+  def checkbox(validation: Option[Modifier] = None, inputId: ComponentId = UdashBootstrap.newId())
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/form/UdashForm.scala]: These helper methods (`*input`, `checkbox` etc.) need better argument documentation with content and input varargs.

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala [hunk 3]
```diff
@@ -2,13 +2,31 @@ package io.udash.bootstrap
...
+  /** Shows the tooltip. */
+  def show(): Unit
+
+  /** Hides the tooltip. */
+  def hide(): Unit
+
+  /** Toggles tooltip visibility. */
+  def toggle(): Unit
+
+  /** Destroys the tooltip. */
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: Needs more details. What changes in its behaviour afterwards? Is it important to destroy those?

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala [hunk 4]
```diff
@@ -26,11 +44,13 @@ trait TooltipUtils[TooltipType <: Listenable[TooltipType, _]] {
   val ManualTrigger = new Trigger("manual")
 
   sealed trait TooltipEvent extends AbstractCase with ListenableEvent[TooltipType]
-  final case class TooltipShowEvent(source: TooltipType) extends TooltipEvent
-  final case class TooltipShownEvent(source: TooltipType) extends TooltipEvent
-  final case class TooltipHideEvent(source: TooltipType) extends TooltipEvent
-  final case class TooltipHiddenEvent(source: TooltipType) extends TooltipEvent
-  final case class TooltipInsertedEvent(source: TooltipType) extends TooltipEvent
+  object TooltipEvent {
+    case class ShowEvent(source: TooltipType) extends TooltipEvent
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: Any reason for these to no longer be final?

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala [hunk 5]
```diff
@@ -66,16 +86,73 @@ trait TooltipUtils[TooltipType <: Listenable[TooltipType, _]] {
...
+           trigger: Seq[Trigger] = defaultTrigger,
+           viewport: Viewport = Viewport("body", 0))
+          (el: dom.Node)
+          (implicit langProperty: LangProperty, translationProvider: TranslationProvider): TooltipType = {
+
+    val tp = apply(animation, container, _ => "", delay, html, placement, template, _ => "", trigger, viewport)(el)
+
+    var lastContent = ""
+    var lastTitle = ""
+    def updateContent(): Unit = {
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: I don't particularly enjoy this style with logic encapsulated in a locally created closure, but there aren't many better ways to do this (unless the `lastContent` and `lastTitle` are useful data for a tooltip user, which they could be - up to you). Can we at least have a test? ;)

Comment (Starzu) [reply bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/TooltipUtils.scala]: It's a hack. I'm not proud of this code, but it works. ;)

File: bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/UdashTooltip.scala [hunk 6]
```diff
@@ -21,14 +21,14 @@ final class UdashTooltip private(selector: UdashTooltip.UdashTooltipJQuery)
   def toggle(): Unit =
     selector.tooltip("toggle")
 
-  /** Destroys the tooltip. */
+  /** Hides and destroys an element's popover.
+    * Check <a href="https://getbootstrap.com/docs/3.3/javascript/#popovers-methods">Bootstrap Docs</a> from more details. */
```
Comment (ddworak) [inline bootstrap/frontend/src/main/scala/io/udash/bootstrap/tooltip/UdashTooltip.scala]: **for** more

**Pull request merged**